### ISSUE TITLE
Provide a way to translate entity references from context

### ISSUE THREAD

**Issue by Leksat** [drupal-graphql/graphql#464]
As mentioned in #463, currently it's only possible to get a translation of a referenced entity  by using the `entityTranslation` field.

One issue here is that it requires `$language` parameter which may be not available on the frontend.

For example, on one of our projects, we know the language prefix from the URL path, but we don't know the language ID which should be passed as `$language`. Yet we prefix API calls with the proper language prefix on this project, so entity repository can `getTranslationFromContext()`.

Here are options to solve this:
1. Use `entityTranslation` field, but make its `$language` argument optional. If it's omitted, use api-level language (`getTranslationFromContext()`).
2. Introduce `entityTranslationFromContext` field.
    - For me this would be a bit more clear way of doing option 1.
3. Change the behavior: return the api-level language by default and use `entityTranslation` field if you need another language.
    - This was proposed by @pmelab.
4. As mentioned in #463 by @fubhy, we can add `$language` argument to `entity` property of entity reference fields. And we can also make it optional, so by default it will use api-level language as suggested in option 3.

**Comment by Leksat** [comment]
As discussed in Slack, I'll implement `entityTranslationFromContext` in a custom module for now.

**Comment by tguilpain** [comment]
@Leksat , were you able to make it work ? Would you share your implementation of entityTranslationFromContext ?
i have been trying to do 4. without much success as of now.

**Comment by Leksat** [comment]
@tguilpain here is the code:
```php
<?php

namespace Drupal\custom_graphql\Plugin\GraphQL\Fields\Entity;

use Drupal\Core\Entity\EntityInterface;
use Drupal\graphql_core\Plugin\GraphQL\Fields\Entity\EntityTranslation;
use Youshido\GraphQL\Execution\ResolveInfo;

/**
 * Retrieve an entity translation.
 *
 * @GraphQLField(
 *   id = "entity_translation_from_context",
 *   secure = true,
 *   name = "entityTranslationFromContext",
 *   nullable = true,
 *   multi = false,
 *   type = "Entity",
 *   parents = {
 *     "Entity",
 *   },
 * )
 */
class EntityTranslationFromContext extends EntityTranslation {

  /**
   * {@inheritdoc}
   */
  public function resolveValues($value, array $args, ResolveInfo $info) {
    if ($value instanceof EntityInterface) {
      yield $this->entityRepository->getTranslationFromContext($value);
    }
  }

}
```
Works fine for me.

**Comment by fubhy** [comment]
I believe this is solved with #509
